strongbox-web-core/src/main/java/org/carlspring/strongbox/web/RepositoryMethodArgumentResolver.java

```diff
@@ -0,0 +1,142 @@
+    public static final String NOT_FOUND_REPOSITORY_MESSAGE = "Could not find requested repository %s:%s.";
+    public static final String NOT_IN_SERVICE_REPOSITORY_MESSAGE = "Requested repository %s:%s is out of service.";
+
+    @Inject
+    protected ConfigurationManager configurationManager;
+
+    @Override
+    public boolean supportsParameter(final MethodParameter parameter)
+    {
+
+        // Check parameter annotation type
+        if (!parameter.hasParameterAnnotation(RepositoryMapping.class))
+        {
+            return false;
+        }
+        // Check parameter type.
+        return parameter.getParameterType().equals(Repository.class);
+    }
+
+    @Override
+    public Object resolveArgument(final MethodParameter parameter,
+                                  final ModelAndViewContainer modelAndViewContainer,
+                                  final NativeWebRequest nativeWebRequest,
+                                  final WebDataBinderFactory webDataBinderFactory)
+            throws MissingPathVariableException
+    {
+        final RepositoryMapping repositoryMapping = parameter.getParameterAnnotation(RepositoryMapping.class);
+        final String storageVariableName = repositoryMapping.storageVariableName();
+        final String storageId = validateAndGetPathVariable(parameter, nativeWebRequest, storageVariableName);
+
+        final String repositoryVariableName = repositoryMapping.repositoryVariableName();
+        final String repositoryId = validateAndGetPathVariable(parameter, nativeWebRequest, repositoryVariableName);
+
+        Repository repository = (Repository) nativeWebRequest.getAttribute(REPOSITORY_REQUEST_ATTRIBUTE,
+                                                                           RequestAttributes.SCOPE_REQUEST);
+
+        if (repository != null && Objects.equals(repository.getId(), repositoryId) &&
+            Objects.equals(repository.getStorage().getId(), storageId))
+        {
+            return repository;
+        }
+
+        final Storage storage = getStorage(storageId);
+        if (storage == null)
+        {
+            final String message = String.format(NOT_FOUND_STORAGE_MESSAGE, storageId);
+            throw new StorageNotFoundException(message);
+        }
+
+        repository = getRepository(storageId, repositoryId);
+        if (repository == null)
+        {
+            final String message = String.format(NOT_FOUND_REPOSITORY_MESSAGE, storageId, repositoryId);
+            throw new RepositoryNotFoundException(message);
+        }
+
+        final boolean inService = repository.isInService();
+        if (!inService)
+        {
+            final String message = String.format(NOT_IN_SERVICE_REPOSITORY_MESSAGE, storageId, repositoryId);
+            throw new ServiceUnavailableException(message);
+        }
+
+        return repository;
+    }
+
+    private String validateAndGetPathVariable(final MethodParameter parameter,
```

Comment:
`getRequiredPathVariable` sounds better here in my opinion than `validateAndGetPathVariable`